runtime/depends_on.cc

```diff
@@ -40,35 +40,40 @@ class dependencies : public recursive_node_visitor {
   depends_on_result* no_dummy(depends_on_result* deps) const { return deps != &dummy_deps ? deps : nullptr; }
 
   void visit(const variable* op) override {
+    if (is_pure && op->field != buffer_field::none) is_pure = false;
     if (depends_on_result* deps = find_deps(op->sym)) {
-      deps->var = true;
+      switch (op->field) {
+      case buffer_field::none: deps->var = true; break;
+      case buffer_field::min:
+      case buffer_field::max:
+        deps->buffer_bounds = true;
+        deps->buffer_dims = true;
+        break;
+      case buffer_field::stride:
+      case buffer_field::fold_factor: deps->buffer_dims = true; break;
```

Comment:
nit: wonky indentation here and below, convention elsewhere seems to be that a multi-case should go onl a newline

Reply:
I agree but this is what clang-format wants to do